src/thread_task.rs

```diff
@@ -659,16 +663,15 @@ unsafe fn delayed_fg_task<'l, 'ttd: 'l>(
         if (row + 1) < progmax {
             ttd.cond.notify_one();
         } else if row + 1 >= progmax {
-            let mut delayed_fg = ttd.delayed_fg.lock().unwrap();
-            delayed_fg.exec = 0 as c_int;
-            *task_thread_lock = Some(delayed_fg);
+            *task_thread_lock = ttd.lock.lock().ok();
+            ttd.delayed_fg_exec.store(0, Ordering::Relaxed);
```

Comment:
I want to replicate the existing locking behavior exactly. This lock might have been intended to lock the exec field, but it might also be for ordering other thread operations. We can revisit this if we ever remove this lock (if we lock data entirely right, which we're not doing yet because of DisjointMut, then we don't need this global thread lock).